### PR TITLE
Add API for grades data

### PR DESCRIPTION
**Pull request by mikix**
Credentials stores a copy of the grades for users in programs with certificates. This adds an API to set that data.

https://openedx.atlassian.net/browse/LEARNER-5051

### CODE REVIEW

File: credentials/apps/core/migrations/0016_admins_can_add_grades.py [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+
+from credentials.apps.core.constants import Role
+
+
+def add_perm(apps, schema_editor):
+    """ Assign the permission."""
+    Group = apps.get_model('auth', 'Group')
+    Permission = apps.get_model('auth', 'Permission')
+
+    group, __ = Group.objects.get_or_create(name=Role.ADMINS)
```
Comment (jlajoie) [inline credentials/apps/core/migrations/0016_admins_can_add_grades.py]: Dumb question, does this being two underscores matter?

File: credentials/apps/records/migrations/0003_grades_username_mode_changes.py [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+            model_name='usergrade',
+            name='username',
+            field=models.CharField(default='', max_length=150),
+            preserve_default=False,
+        ),
+        migrations.AlterUniqueTogether(
+            name='usergrade',
+            unique_together=set([('username', 'course_run')]),
+        ),
+        migrations.RemoveField(
```
Comment (jlajoie) [inline credentials/apps/records/migrations/0003_grades_username_mode_changes.py]: With removing fields I think there's some special stuff we need to do to support our blue/green deploys, e.g. deleting all of the references to the columns we plan to delete first, then dropping in a subsequent PR the actual columns.

File: credentials/apps/core/migrations/0016_admins_can_add_grades.py (hunk 1 above)
Comment (mikix) [reply credentials/apps/core/migrations/0016_admins_can_add_grades.py]: Hmm, no. I think that was just from some copied migration code that did that. I'll drop to one.

File: credentials/apps/records/migrations/0003_grades_username_mode_changes.py (hunk 2 above)
Comment (mikix) [reply credentials/apps/records/migrations/0003_grades_username_mode_changes.py]: Ah yes, true. Thankfully for this model, no code references it at all yet.

File: credentials/apps/core/migrations/0016_admins_can_add_grades.py (hunk 1 above)
Comment (mikix) [reply credentials/apps/core/migrations/0016_admins_can_add_grades.py]: Done.

File: credentials/apps/core/migrations/0016_admins_can_add_grades.py [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+    group.permissions.add(permission)
+
+
+def remove_perm(apps, schema_editor):
+    """ Remove the permission."""
+    Group = apps.get_model('auth', 'Group')
+    Permission = apps.get_model('auth', 'Permission')
+
+    group, __ = Group.objects.get_or_create(name=Role.ADMINS)
+    permission = Permission.objects.get(codename='add_usergrade')
```
[the remark this answers is not in the document]
Comment (mikix) [reply credentials/apps/core/migrations/0016_admins_can_add_grades.py]: Done.

**Pull request merged**